src/coreclr/jit/objectalloc.cpp

```diff
@@ -828,7 +828,7 @@ unsigned int ObjectAllocator::MorphNewArrNodeIntoStackAlloc(GenTreeCall*
         blockSize = AlignUp(blockSize, 8);
     }
 
-    comp->lvaSetStruct(lclNum, comp->typGetBlkLayout(blockSize), /* unsafeValueClsCheck */ false);
+    comp->lvaSetStruct(lclNum, comp->typGetArrayLayout(clsHnd, length), /* unsafe */ false);
```

Comment:
Would it make sense to propagate the layout through instead of `blockSize`? We could probably save a call to `getChildType`.

Reply:
Yeah, I should get rid of the redundant layout fetch.

Reply:
Only a small step from there to using layouts everywhere. 

I think I should do that first (leaving GC arrays disabled) and verify zero diff, then come back to this and enable gc arrays.

Reply:
After poking at it some, I think it might be simpler to take this change first and then revise the entire thing to be more layout centric (getting rid of the special box type, etc).

Thoughts?

Reply:
Sounds fine to me.